engine/src/main/scala/cromwell/engine/workflow/tokens/JobTokenDispenserActor.scala

```diff
@@ -120,20 +121,32 @@ class JobTokenDispenserActor(override val serviceRegistryActor: ActorRef,
     case ShutdownCommand => context stop self
   }
 
+  // If GroupMetricsActor is defined, periodically refresh our list of quota-exhausted groups.
+  // We'll use this list to prevent giving new execution tokens to groups that can't use them due
+  // to already being limited by quota.
+  private def quotaExhaustedGroupsReceive: Receive = {
+    case RefreshQuotaExhaustedGroups(interval) =>
+      groupMetricsActor.foreach(gmActor => gmActor ! GetQuotaExhaustedGroups)
+      context.system.scheduler.scheduleOnce(interval)(self ! RefreshQuotaExhaustedGroups(interval))(
+        context.dispatcher
+      )
+      ()
+    case GetQuotaExhaustedGroupsSuccess(quotaExhaustedGroups) =>
+      this.quotaExhaustedGroups = quotaExhaustedGroups
+    case GetQuotaExhaustedGroupsFailure(errorMsg) =>
+      log.error(s"Failed to fetch quota exhausted groups. Error: $errorMsg")
+      this.quotaExhaustedGroups = List.empty // is this the behavior we want?
```

Comment:
I think no groups exhausted is the fail-safe choice. If the metrics system gets messed up, work can still run.

Reply:
I agree. This makes sense. Plus it is behaving same as before where we used empty list when there was any sort of failure in fetching quota exhausted groups (previous lines L156-L161)